error.go

```diff
@@ -54,6 +55,113 @@ func RootCause(err error) error {
 	}
 }
 
+// errf is a version of fmt.Errorf with support for a chain of multiple
```

Comment:
From the documentation at least, this seems like `errors.Wrapf` with a slightly tweaked API: 

- You can bundle multiple errors at 1 callsite: why would we want to do this? I can't easily imagine a scenario where multiple errors need to be nested at once.

- The `error` argument is optional unlike Wrapf. This seems more ergonomic if we're actually expecting _large_ error chains but still, easy enough to work around.

Reply:
Ah, It looks like I didn't mention in the PR why we want to make error
chains like this. I'll add it to the PR message but:

We want to chain errors where one portion of a long error message
reasonably ends, so that it chains into the next one. Using the chains
as markers for separate sentences, we'll be able to split the error
messages across multiple lines so that dig stops printing massive walls
of text.

Reply:
@mh-park @vyshah added more to the PR description